### PR TITLE
add setting policy for post-install updates via autoinstall

### PR DESCRIPTION
**Pull request by dbungert**
This can be controlled with autoinstall
updates: security or all

Also an API for controlling this:
curl --silent --unix-socket .subiquity/socket a/updates ->
"security"
curl -d '"all"' --unix-socket .subiquity/socket a/updates

### CODE REVIEW

File: subiquity/models/subiquity.py [hunk 1]
```diff
@@ -81,6 +82,7 @@ def setup_yaml():
     "packages",
     "snaplist",
     "ssh",
+    # "updates", # FIXME hey mwhudson - do I need this?
```
Comment (mwhudson) [inline subiquity/models/subiquity.py:85]: Not unless you want to force the client to provide a value for this -- which I don't think we do.

File: subiquity/common/apidef.py [hunk 2]
```diff
@@ -44,11 +44,12 @@
...
 class API:
     """The API offered by the subiquity installer process."""
+    identity = simple_endpoint(IdentityData)
     locale = simple_endpoint(str)
-    proxy = simple_endpoint(str)
     mirror = simple_endpoint(str)
-    identity = simple_endpoint(IdentityData)
+    proxy = simple_endpoint(str)
     ssh = simple_endpoint(SSHData)
+    updates = simple_endpoint(str)
```
Comment (mwhudson) [inline subiquity/common/apidef.py:52]: I'd ever so slightly prefer to have this API method take an enum rather than a string.

**Review by mwhudson — COMMENTED** [review]
Hey, this looks pretty good! Just a couple of comments. I guess we should have a conversation about whether we really do want to allow autoinstalls to disable security updates.

File: subiquity/server/controllers/updates.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        self.model.updates = data
+
+    def make_autoinstall(self):
+        return self.model.updates
+
+    def serialize(self):
+        return self.model.updates
+
+    def deserialize(self, data):
+        self.model.updates = data
```
Comment (mwhudson) [inline subiquity/server/controllers/updates.py:51]: oh, forgot: you don't need these methods

Comment (mwhudson) [reply subiquity/server/controllers/updates.py:51]: oh actually maybe it makes sense to keep them. i'm sure I'll have made up my mind by tomorrow :)

File: subiquity/models/subiquity.py (hunk 1 above)
Comment (dbungert) [reply subiquity/models/subiquity.py:85]: Agree, removing this line

File: subiquity/server/controllers/updates.py (hunk 3 above)
Comment (dbungert) [reply subiquity/server/controllers/updates.py]: SubiquityController wants deserialize.  serialize might be not required at the API level.

File: subiquity/common/apidef.py (hunk 2 above)
Comment (dbungert) [reply subiquity/common/apidef.py:52]: I messed with putting enum on the API boundary and it made a mess of things.  I'm willing to try again but at a certain point I cut my losses and pushed a simpler version.

**Review by mwhudson — APPROVED** [review]
nice

**Pull request merged**